src/controllers/user.controller.js

```diff
@@ -1,11 +1,133 @@
 import checkEmailpassword from '../middlewares/user.middleware';
 import UserServices from '../services/user.service';
-import response from '../helpers/response.helper';
+import EncryptPassword from '../helpers/Encryptor';
+import Token from '../helpers/token';
+import response from '../helpers/response.handler';
+import sendEmail from '../helpers/send.email.helper';
 
 /**
  * Class for users related operations such Sign UP, Sign In and others
  */
 class userController {
+/**
+   * signup a user and saving user data in the database
+   * @param {Object} req The request object
+   * @param {Object} res The response object
+   * @returns {Object} A user object with selected fields
+   */
+  static async signup(req, res) {
+    try {
+      const {
+        firstName,
+        lastName,
+        email,
+        gender,
+        country,
+        birthday,
+        phoneNumber
+      } = req.body;
+      const password = await EncryptPassword(req.body.password);
+      const isVerified = false;
+      const token = Token.GenerateToken(email, password, isVerified, firstName);
+      const NewUser = {
+        firstName,
+        lastName,
+        email,
+        gender,
+        country,
+        birthday,
+        password,
+        phoneNumber,
+        isVerified: false,
+        token
+      };
+      UserServices.CreateUser(NewUser);
+      const data = {
+        token,
+      };
+      const userName = firstName;
+      const view = `<!DOCTYPE html>
```

Comment:
I think it would be nice to set create email templates in another folder.

Reply:
Thanks,I solved it !